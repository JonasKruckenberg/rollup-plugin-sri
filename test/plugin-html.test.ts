import { getHtml } from './util'
import { join } from 'path'
import { rollup } from 'rollup'
import html from '@rollup/plugin-html'
import sri from '../index'

process.chdir(join(__dirname, 'fixtures'));

describe('@rollup/plugin-html', () => {
    test('basic', async () => {
        const bundle = await rollup({
            input: '1.js',
            plugins: [
                html({}),
                sri()
            ]
        })

        const source = await getHtml(bundle)
        expect(source).toMatchSnapshot()
    })
    test('public path', async () => {
        const bundle = await rollup({
            input: '1.js',
            plugins: [
                html({
                    publicPath: '/assets'
                }),
                sri({ 
                    publicPath:'/assets' 
                })
            ]
        })

        const source = await getHtml(bundle)
        expect(source).toMatchSnapshot()
    })
})