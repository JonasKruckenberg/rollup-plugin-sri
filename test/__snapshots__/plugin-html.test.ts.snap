// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@rollup/plugin-html basic 1`] = `
Object {
  "fileName": "index.html",
  "source": "<!DOCTYPE html><html lang=\\"en\\"><head>
    <meta charset=\\"utf-8\\">
    <title>Rollup Bundle</title>
    
  </head>
  <body>
    <script src=\\"1.js\\" type=\\"module\\" integrity=\\"sha384-hxLOTPNeroviduGwj6Po7ksKHF0YRfwVZGjcsAnyJGlTHnQFW/268mrNS7EhSMvT\\" crossorigin=\\"anonymous\\"></script>
  
</body></html>",
}
`;

exports[`@rollup/plugin-html public path 1`] = `
Object {
  "fileName": "index.html",
  "source": "<!DOCTYPE html><html lang=\\"en\\"><head>
    <meta charset=\\"utf-8\\">
    <title>Rollup Bundle</title>
    
  </head>
  <body>
    <script src=\\"/assets1.js\\" type=\\"module\\" integrity=\\"sha384-hxLOTPNeroviduGwj6Po7ksKHF0YRfwVZGjcsAnyJGlTHnQFW/268mrNS7EhSMvT\\" crossorigin=\\"anonymous\\"></script>
  
</body></html>",
}
`;
